docs/requirements.txt

```
@@ -0,0 +1,5 @@
+git+https://github.com/SymbiFlow/sphinx_symbiflow_theme
+
+sphinx
+symbolator
+sphinxcontrib-verilog-diagrams
```

This should reference the local package, otherwise it will be downloaded from PyPi?
```suggestion
-e ../
```

---

This is about `sphinxcontrib-verilog-diagrams` not about `sphinx-verilog-domain`.

Nevertheless, the relative dependencies are not handled properly by conda. If you add the `-e ../` to the requirements file, the main environment (created by make-env) will not be created. Conda seems to handle relative dependencies in the requirements file, as a relative to the environment.yml not to the requirements file itself:

```
Pip subprocess error:
ERROR: File "setup.py" not found. Directory cannot be installed in editable mode: /mnt/hdd1//verilog_domain_ws
```

The repository is placed in:
```
/mnt/hdd1/verilog_domain_ws/sphinx-verilog-domain
```

The `sphinx-verilog-domain` package is installed in the `environment.yml` directly.
Since the RTD build also uses conda and the documentation build should happen inside this environment, I think that the `-e ../` can be omitted in the `docs/requirements.txt`

---

@mithro do you agree to left those files in that way? The problem with this convention will never occur unless somebody will decide to build the documentation without using `make-env` inside this repository. This is exactly the same way as it is done for `sphinxcontrib-verilog-diagrams`.

---

@rw1nkler - I was sure that I had solved this somewhere...

---

There is a bunch of discussion at https://github.com/pypa/pip/issues/328

---

It seems that this problem is not resolved for the `requirements.txt` (It will be fixed by the pipfile):
- https://github.com/pypa/pip/issues/6112
- https://github.com/pypa/pip/issues/1795

As I said before, the problem with the current settings will happen only if somebody will build the docs outside the make-env environment, which is generally wrong. I think that we should leave the requirements file as they are now.